core/src/main/java/io/opencensus/stats/MeasureMap.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import io.opencensus.stats.Measure.DoubleMeasure;
+import io.opencensus.stats.Measure.LongMeasure;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * A map from {@link Measure}'s to measured values.
+ */
+public final class MeasureMap implements Iterable<Measurement> {
+  /**
+   * Constructs a {@link MeasureMap} from the given {@link DoubleMeasure}
+   * and associated value.
+   */
+  public static MeasureMap of(DoubleMeasure measure, double value) {
+    return builder().put(measure, value).build();
+  }
+
+  /**
+   * Constructs a {@link MeasureMap} from the given {@link DoubleMeasure}'s
+   * and associated values.
+   */
+  public static MeasureMap of(DoubleMeasure measure1, double value1,
+      DoubleMeasure measure2, double value2) {
```

Comment:
I agree, and I think now that we have different Measure types, maybe it's better to replace the static method `of()` with `with()`, similar to [StatsContext.with()](https://github.com/census-instrumentation/opencensus-java/blob/master/core/src/main/java/io/opencensus/stats/StatsContext.java#L31).